sklearn/utils/estimator_checks.py

```diff
@@ -360,55 +366,242 @@ def check_transformer_pickle(name, Transformer):
     assert_array_almost_equal(pickled_X_pred, X_pred)
 
 
-def check_estimators_partial_fit_n_features(name, Alg):
-    # check if number of features changes between calls to partial_fit.
-    if not hasattr(Alg, 'partial_fit'):
+def check_partial_fit_n_features(name, Estimator):
+    """Check if ValueError is raised if number of features changes between
+    multiple partial_fit calls.
+    """
+
+    if not hasattr(Estimator, 'partial_fit'):
         return
+
     X, y = make_blobs(n_samples=50, random_state=1)
     X -= X.min()
+
     with warnings.catch_warnings(record=True):
-        alg = Alg()
-    set_fast_parameters(alg)
-    if isinstance(alg, ClassifierMixin):
+        estimator = Estimator()
+
+    set_random_state(estimator)
+    set_fast_parameters(estimator)
+
+    if isinstance(estimator, ClassifierMixin):
         classes = np.unique(y)
-        alg.partial_fit(X, y, classes=classes)
+        estimator.partial_fit(X, y, classes=classes)
+        assert_raises(ValueError, estimator.partial_fit, X[:, :-1], y)
+
+    elif isinstance(estimator, RegressorMixin):
```

Comment:
you can just do ``else`` here. All transformers take an optional ``y`` that might be ignored.